src/EditorFeatures/Core/Implementation/IntelliSense/AsyncCompletion/CommitManager.cs

```diff
@@ -174,6 +178,17 @@ internal CommitManager(ITextView textView, RecentItemsManager recentItemsManager
                 return AsyncCompletionData.CommitBehavior.None;
             }
 
+            if (view.Properties.TryGetProperty(CompletionSource.TargetTypeFilterExperimentEnabled, out bool isExperimentEnabled) && isExperimentEnabled)
+            {
+                // Capture the % of committed completion items that would have appeared in the "Target type matches" filter
+                // (regardless of whether that filter button was active at the time of commit).
+                Logger.Log(FunctionId.Intellisense_AsyncCompletion_CommitWithTargetTypeCompletionExperimentEnabled);
+                if (item.Filters.Any(f => f.DisplayText == FeaturesResources.Target_type_matches))
+                {
+                    Logger.Log(FunctionId.Intellisense_AsyncCompletion_CommitItemWithTargetTypeFilter);
+                }
+            }
```

Comment:
We can move this code into TryCommit. Passing VSCompletionItem here seems to be unnecessary.